### PR TITLE
PP-9113 payment api representation 

### PR DESCRIPTION
**Pull request by gidsg**
Transforms ledger transaction to produce API representation of a payment (excluding the _links block)

N.B. This leaves a TODO for Refunds which is assumed will require a separate transformation.

### CODE REVIEW

File: src/main/resources/config/config.yaml [hunk 1]
```diff
@@ -28,7 +28,6 @@ database:
   properties:
     charSet: UTF-8
     hibernate.dialect: org.hibernate.dialect.PostgreSQLDialect
-    hibernate.types.print.banner: false
```
Comment (alexbishop1) [inline src/main/resources/config/config.yaml:31]: Why was this removed?

**Review by alexbishop1 — APPROVED** [review]
Approving this because there’s nothing wrong with it but there’s a few suggestions you may wish to consider.

File: src/main/java/uk/gov/pay/webhooks/message/WebhookMessageService.java [hunk 2]
```diff
@@ -69,15 +69,18 @@ public void handleInternalEvent(InternalEvent event) throws IOException, Interru
     }
 
     private WebhookMessageEntity buildWebhookMessage(WebhookEntity webhook, InternalEvent event, LedgerTransaction ledgerTransaction) {
-        JsonNode resource = objectMapper.valueToTree(ledgerTransaction); 
-
+        JsonNode resource = switch (event.resourceType()) {
+            case "payment" -> objectMapper.valueToTree(PaymentApiRepresentation.of(ledgerTransaction));
+//          TODO: Add refund case transformation
+            default -> objectMapper.valueToTree(ledgerTransaction);
+        };
```
[the remark this answers is not in the document]
Comment (gidsg) [reply src/main/java/uk/gov/pay/webhooks/message/WebhookMessageService.java:76]: No, used it here first! https://github.com/alphagov/pay-webhooks/blob/2e859ee8d9297fde220efa5f1f357310536bb30f/src/main/java/uk/gov/pay/webhooks/webhook/WebhookService.java#L112-L122

File: src/main/java/uk/gov/pay/webhooks/message/apirepresentation/PaymentApiAddress.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package uk.gov.pay.webhooks.message.apirepresentation;
+
+public record PaymentApiAddress(String line1,
+                                String line2,
+                                String postcode,
+                                String city,
+                                String country) {
```
[the remark this answers is not in the document]
Comment (gidsg) [reply src/main/java/uk/gov/pay/webhooks/message/apirepresentation/PaymentApiAddress.java:7]: Can do, will have to override the auto-serialised value which maybe makes it a bit less clean.

[the remark this answers is not in the document]
Comment (alexbishop1) [reply src/main/java/uk/gov/pay/webhooks/message/apirepresentation/PaymentApiAddress.java:7]: Yes but it should be possible to just use `ToStringSerializer`.

File: src/main/java/uk/gov/pay/webhooks/message/apirepresentation/PaymentApiRepresentation.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+public record PaymentApiRepresentation(
+        String paymentId,
+        Long amount,
+        PaymentState state,
+        String returnUrl,
+        String description,
+        String reference,
+        String email,
+        String paymentProvider,
+        String createdDate,
```
[the remark this answers is not in the document]
Comment (gidsg) [reply src/main/java/uk/gov/pay/webhooks/message/apirepresentation/PaymentApiRepresentation.java:25]: Yes, I agree this is a clunky type, but this is how it comes from the LedgerTransaction model: https://github.com/alphagov/pay-webhooks/blob/e098b1e95b7f5df1d3abdeadc26255ae2457a61e/src/main/java/uk/gov/pay/webhooks/ledger/model/LedgerTransaction.java#L126-L128 so seems weird to transform it for no reason?

File: src/main/resources/config/config.yaml (hunk 1 above)
Comment (gidsg) [reply src/main/resources/config/config.yaml:31]: I found the banner was still showing up for tests, this PR adds it instead to `src/main/resources/hibernate.properties` which seems to remove it everywhere.

File: src/main/java/uk/gov/pay/webhooks/message/apirepresentation/PaymentApiRepresentation.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (alexbishop1) [reply src/main/java/uk/gov/pay/webhooks/message/apirepresentation/PaymentApiRepresentation.java:25]: I see your point: it comes in as a string, it goes out as a string. Part of me wishes it would be of an appropriate type when it’s actually inside the microservice, so it’s at least pure and correct during the times it doesn’t have to interface with the real world. So I won’t complain if you leave it as a string.

**Comment by gidsg** [comment]
@alexbishop1 Thanks, I've implemented a couple of your suggestions, please may you re-review?

**Review by alexbishop1 — APPROVED** [review]
